paimon-core/src/main/java/org/apache/paimon/sort/BinaryExternalSortBuffer.java

```diff
@@ -120,7 +120,7 @@ public static BinaryExternalSortBuffer create(
             int maxNumFileHandles,
             CompressOptions compression,
             MemorySize maxDiskSize) {
-        RecordComparator comparator = newRecordComparator(rowType.getFieldTypes(), keyFields);
+        RecordComparator comparator = newRecordComparator(rowType.getFieldTypes(), keyFields, true);
```

Comment:
Why here pass true?

Reply:
Yes, here should keep the same with options. Had change it.